lib/ipam/ipam.go

```diff
@@ -205,42 +206,76 @@ func (c ipamClient) getBlockFromAffinity(ctx context.Context, aff *model.KVPair)
 // determinePools compares a list of requested pools with the enabled pools
 // and returns the intersect. If any requested pool does not exist, or is not enabled, an error is returned.
 // If no pools are requested, all enabled pools are returned.
-func (c ipamClient) determinePools(requestedPoolNets []net.IPNet, version int) ([]v3.IPPool, error) {
+func (c ipamClient) determinePools(requestedPoolNets []net.IPNet, version int, node *v3.Node) ([]v3.IPPool, error) {
+	// Nil check input node object.
```

Comment:
So, if you're going to assert this is never nil, then why make it a pointer in the first place?

Strictly speaking, this function only needs a node passed in if an enabled IP pool has a node selector.

Reply:
I agree that the node _should_ only be passed in if an enabled IP pool has a node selector, but the fact that this method actually fetches the pool makes this a chicken/egg problem. Would prefer to provide the egg, even if we don't end up using it..

Reply:
It looks like calico Nodes are created as a pointer to the Node object. Not sure how this is stored/retrieved by the backing datastore though...
https://github.com/projectcalico/node/blob/master/pkg/startup/startup.go#L345